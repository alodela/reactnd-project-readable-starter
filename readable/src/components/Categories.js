import React, { Component } from 'react'
import { Nav, NavItem, NavLink } from 'reactstrap'
import Api from '../utils/api'

class Categories extends Component {
    state = {
        categories: []
    };

    componentDidMount() {
        Api.fetchCategories()
            .then((categories) => this.setState({ categories: categories }))
    }

    render() {
        const { categories } = this.state

        return (
            <div>
                <h3>Categories</h3>
                <hr />
                <Nav vertical>
                    <NavItem>
                        <NavLink href="/">All</NavLink>
                    </NavItem>
                    {
                        categories.map((category) => {
                            return (
                                <NavItem key={category.name}>
                                    <NavLink href={`/categories/${category.path}/posts`}>{category.name}</NavLink>
                                </NavItem>
                            );
                        })
                    }
                </Nav>
            </div>
        )
    }
}

export default Categories