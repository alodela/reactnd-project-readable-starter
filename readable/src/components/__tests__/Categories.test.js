import React from 'react';
import { shallow } from 'enzyme';
import Categories from './../Categories';

const CATEGORIES = [
	{ name: 'React', path: 'react' },
	{ name: 'Redux', path: 'redux' }
];

describe('Categories component', () => {
    it('should render correctly', () => {
        const wrapper = shallow(
            <Categories categories={CATEGORIES} />
        );

        expect(wrapper).toMatchSnapshot();
    });
});