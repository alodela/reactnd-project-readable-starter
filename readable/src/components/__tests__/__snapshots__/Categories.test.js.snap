// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories component should render correctly 1`] = `
<div>
  <h3>
    Categories
  </h3>
  <hr />
  <Nav
    tag="ul"
    vertical={true}
  >
    <NavItem
      tag="li"
    >
      <NavLink
        href="/"
        tag="a"
      >
        All
      </NavLink>
    </NavItem>
    <NavItem
      key="React"
      tag="li"
    >
      <NavLink
        href="/categories/react/posts"
        tag="a"
      >
        React
      </NavLink>
    </NavItem>
    <NavItem
      key="Redux"
      tag="li"
    >
      <NavLink
        href="/categories/redux/posts"
        tag="a"
      >
        Redux
      </NavLink>
    </NavItem>
  </Nav>
</div>
`;
